analyzers/src/SonarAnalyzer.CSharp/Rules/Hotspots/ClearTextProtocolsAreSensitive.cs

```diff
@@ -81,23 +81,27 @@ public sealed class ClearTextProtocolsAreSensitive : HotspotDiagnosticAnalyzer
             KnownType.System_Windows_Markup_XmlnsCompatibleWithAttribute,
         };
 
-        private readonly CSharpObjectInitializationTracker objectInitializationTracker =
+        private static readonly CSharpObjectInitializationTracker ObjectInitializationTracker =
             new(constantValue => constantValue is bool value && value,
                                                   ImmutableArray.Create(KnownType.System_Net_Mail_SmtpClient, KnownType.System_Net_FtpWebRequest),
                                                   propertyName => propertyName == EnableSslName);
 
-        private readonly Regex httpRegex;
-        private readonly Regex ftpRegex;
-        private readonly Regex telnetRegex;
-        private readonly Regex telnetRegexForIdentifier;
-        private readonly Regex validServerRegex;
+        private static readonly Regex HttpRegex;
+        private static readonly Regex FtpRegex;
+        private static readonly Regex TelnetRegex;
+        private static readonly Regex TelnetRegexForIdentifier;
+        private static readonly Regex ValidServerRegex;
 
         public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics =>
             ImmutableArray.Create(DefaultRule, EnableSslRule);
 
         public ClearTextProtocolsAreSensitive() : this(AnalyzerConfiguration.Hotspot) { }
 
         public ClearTextProtocolsAreSensitive(IAnalyzerConfiguration analyzerConfiguration) : base(analyzerConfiguration)
+        {
+        }
+
+        static ClearTextProtocolsAreSensitive()
```

Comment:
This would work (even without the Lazy). I explicitly did not do it, as re-ordering the members would break the logic at runtime. Reordering in the static constructor breaks at compile time.

See also this comment for the same problem but for object initializers:

> All that said, it really shouldn't matter. If you have a class where the order of assignments to two or more properties affects the final outcome, you've got a pretty hazardous class on your hands. Such design ought to be avoided at all costs, as it is way too easy to break in any case, even when the compiler rules for object initialization are reasonably clear.

https://stackoverflow.com/a/47959273

Or this one by John Skeet:

> I would also be very wary of actually writing code which depends on this..

https://stackoverflow.com/a/495622